### PR TITLE
feat: Added a spec to describe the type casting we recommend to do on behalf of the user

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Overview/Summary

- Added a spec to describe the type casting we recommend to do on behalf of the user
- Added an example for a boolean parameter & an int array parameter
image

### CODE REVIEW

**Comment by jtracey93** [comment]
@AlexanderSehr can you resolve the conflicts here please :)

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr can you resolve the conflicts here please :)

Done @jtracey93

File: docs/content/specs/bicep/_index.md [hunk 1]
```diff
@@ -583,3 +583,95 @@ To leverage the feature you must
 ---
 
 <br>
+
+#### ID: BCPNFR17 - Category: Composition - Code Styling - Type casting
+
+To improve the usability of primitive module properties declared as strings, you should declare them as the type they are and apply any required casting in the module on behalf of the user.
```
Comment (eriqua) [inline docs/content/specs/bicep/_index.md:589]: Although I know what the goal is, I'm not sure if "as the type they are" clarifies. I don't have a good suggestion either, but trying: 
```suggestion
To improve the usability of primitive module properties declared as strings, you should declare them using a type which better represents them, and apply any required casting in the module on behalf of the user.
```

File: docs/content/specs/bicep/_index.md [hunk 2]
```diff
@@ -583,3 +583,95 @@ To leverage the feature you must
...
+  {{< /tab >}}
+{{< /tabs >}}
+
+
+<h5>Integer Array as String Array</h5>
+
+{{< tabs "intArrayString" >}}
+  {{< tab "Before" >}}
+
+  ```bicep
```
Comment (eriqua) [inline docs/content/specs/bicep/_index.md:635]: Should this example change based on #817 ?
Also why `zonesValue` instead of `zones`?

Comment (AlexanderSehr) [reply docs/content/specs/bicep/_index.md:635]: It should yes, @eriqua. Updated it. 

I think I called it `zoneValues` to make it explicit that I'm not refering to a property or something. Anyways, changed it to `zones`.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**